### ISSUE TITLE
undefined-loop-variable raised after for x in range(number) where x will always be defined before exiting

### ISSUE THREAD

**Issue by cnjo028** [PyCQA/pylint#2533]
### Steps to reproduce
1. Run this python script
```
def funkier():
    for val in range(3):
        pass
    print(val)
```
2. Pylint output is
```
No config file found, using default configuration
************* Module test
C:  4, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
C:  1, 0: Missing module docstring (missing-docstring)
C:  1, 0: Missing function docstring (missing-docstring)
W:  4,10: Using possibly undefined loop variable 'val' (undefined-loop-variable)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)

```
### Current behavior
Pylint thinks that `val in range( number )` may not get defined

### Expected behavior
`range( number )` will always pass back a value, flag shouldn't raise

### pylint --version output
```
No config file found, using default configuration
pylint 1.9.3, 
astroid 1.6.5
Python 2.7.15 (default, May 16 2018, 17:50:09) 
[GCC 8.1.1 20180502 (Red Hat 8.1.1-1)]
```

**Comment by PCManticore** [comment]
Thank you, should be fixed now.
